js/src/components/paid-ads/asset-group/final-url-card.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { useState } from '@wordpress/element';
+import { ExternalLink } from 'extracted/@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import Section from '.~/wcdl/section';
+import AccountCard, { APPEARANCE } from '.~/components/account-card';
+import AppButton from '.~/components/app-button';
+import AssetsLoader from './assets-loader';
+import './final-url-card.scss';
+
+/**
+ * @typedef {import('.~/data/types.js').AssetGroup} AssetGroup
+ */
+
+/**
+ * Renders the Card UI for managing the final URL and getting the suggested assets.
+ *
+ * @param {Object} props React props.
+ * @param {(assetGroup: AssetGroup | null) => void} props.onAssetsChange Callback function when the suggested assets are changed or reset to `null`.
+ * @param {string} [props.initialFinalUrl] The initial final URL.
+ * @param {boolean} [props.hideFooter=false] Whether to hide the card footer.
+ */
+export default function FinalUrlCard( {
+	onAssetsChange,
+	initialFinalUrl,
+	hideFooter = false,
+} ) {
+	const [ finalUrl, setFinalUrl ] = useState( initialFinalUrl || null );
```

Comment:
💅  This `| null` could be omitted I guess. `undefined` will have the same result.

Reply:
The intention is to align with the state reset below:

https://github.com/woocommerce/google-listings-and-ads/blob/f084e1a13d30b8243e8f19de624fca7b54993540/js/src/components/paid-ads/asset-group/final-url-card.js#L52

So it won't have three possible types: `string`, `null`, and `undefined`.

Of course, it's possible to change the type possibilities to `string` and `undefined`, but the semantics of `setFinalUrl();` and `setFinalUrl( undefined );` are relatively unclear.